### PR TITLE
Implemented C++ with CPM.cmake example

### PR DESCRIPTION
**Pull request by leozz37**
### C++ Example

Implemented C++ example using CMake and [CPM.cmake](https://github.com/TheLartians/CPM.cmake) example.

### CODE REVIEW

**Comment by WSLUser** [comment]
Can you add STL/MSBuild as well?

**Comment by AraHaan** [comment]
Ye MSBuild is used by many that use sln and vc[x]proj files.

**Comment by Phantsure** [comment]
@leozz37 Can you please address the conflicts?

File: README.md [hunk 1]
```diff
@@ -87,6 +87,7 @@ Every programming language and framework has its own way of caching.
 
 See [Examples](examples.md) for a list of `actions/cache` implementations for use with:
 
+- [C++ - CPM.cmake](./examples.md#c---cpm.cmake)
```
Comment (Phantsure) [inline README.md:90]:
```suggestion
- [C++ - CPM.cmake](./examples.md#c---cpmcmake)
```

File: examples.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # Examples
 
 - [Examples](#examples)
+  - [C++ - CPM.cmake](#c---cpm.cmake)
```
Comment (Phantsure) [inline examples.md:4]:
```suggestion
  - [C++ - CPM.cmake](#c---cpmcmake)
```

**Comment by Phantsure** [comment]
@leozz37 Closing this due to no response. You are welcome to re-open it

**Pull request closed without merging**